hastexo-guacamole-client/guacamole/client.py

```diff
@@ -0,0 +1,226 @@
+"""
+The MIT License (MIT)
+
+Copyright (c)   2014 rescale
+                2014 - 2016 Mohab Usama
+"""
+
+import socket
+import logging
+
+from .exceptions import GuacamoleError
+
+from .instruction import INST_TERM
+from .instruction import GuacamoleInstruction as Instruction
+
+# supported protocols
+PROTOCOLS = ('vnc', 'rdp', 'ssh')
+
+PROTOCOL_NAME = 'guacamole'
+
+BUF_LEN = 4096
+
+guac_logger = logging.getLogger(__name__)
+guac_logger.setLevel(logging.INFO)
+guac_logger.handlers = [logging.StreamHandler()]
+
+
+class GuacamoleClient(object):
+    """Guacamole Client class."""
+
+    def __init__(self, host, port, timeout=20, debug=True, logger=None):
```

Comment:
Rather than having a boolean that sets `logging.DEBUG`, I usually prefer to have a numeric parameter that sets the log level, as in:

```python
def __init__(self, host, port, timeout=20, loglevel=logging.DEBUG, logger=None):
```